src/lib/y2firewall/clients/auto.rb

```diff
@@ -198,6 +206,33 @@ def imported
       def imported?
         !!self.class.imported
       end
+
+
+      FIREWALLD_ATTRS = [
```

Comment:
It seems too complex for me:

Once the configuration has been read you don't need to use the `api` and will be more slow, and also the definition of the method to just one method in common is too much from my POV.

```ruby
firewalld.zones.each do |zone|
  summary << "<h2>zone.name</h2>"
  zone.relations.each do |relation|
    summary << "<p>#{relation.to_s.capitalize}: zone.public_send(relation).join(" ")</p>"
  end
```

Reply:
BTW my minimal initial example was something like:

```ruby
summary << "<ul>"
summary << "<li><b>Default ZONE:</b> #{firewalld.default_zone}</li>"
summary << "<li><b>Log denied packets:</b> #{firewalld.log_denied_packets}</li>"
summary << "</ul>"
firewalld.zones.each do |z|
  summary << "<h2>#{z.name}</h2>"
  summary << "<p><b>Interfaces:</b> #{z.interfaces.join(',')}</p>"
  summary << "<p><b>Services:</b> #{z.services.join(',')}</p>"
  summary << "<p><b>Ports:</b> #{z.ports.join(',')}</p>"
  summary << "<p><b>Protocols:</b> #{z.protocols.join(',')}</p>"
  summary << "<p><b>Masquerade:</b> #{z.masquerade?}</p>"
end
```

Reply:
I'll switch to upper layers soon. I've been focusing on layout so far.